vochain/scrutinizer/types/types.go

```diff
@@ -0,0 +1,171 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+	"time"
+
+	tmtypes "github.com/tendermint/tendermint/types"
+	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+// Process represents an election process handled by the Vochain.
+// The scrutinizer Process data type is different from the vochain state data type
+// since it is optimized for querying purposes and not for keeping a shared consensus state.
+type Process struct {
+	ID            types.HexBytes             `badgerholdKey:"ID" json:"processId"`
+	EntityID      types.HexBytes             `badgerholdIndex:"EntityID" json:"entityId"`
+	StartBlock    uint32                     `json:"startBlock"`
+	EndBlock      uint32                     `badgerholdIndex:"EndBlock" json:"endBlock"`
+	Rheight       uint32                     `badgerholdIndex:"Rheight" json:"-"`
+	CensusRoot    types.HexBytes             `json:"censusRoot"`
+	CensusURI     string                     `json:"censusURI"`
+	CensusOrigin  int32                      `json:"censusOrigin"`
+	Status        int32                      `badgerholdIndex:"Status" json:"status"`
+	Namespace     uint32                     `badgerholdIndex:"Namespace" json:"namespace"`
+	Envelope      *models.EnvelopeType       `json:"envelopeType"`
+	Mode          *models.ProcessMode        `json:"processMode"`
+	VoteOpts      *models.ProcessVoteOptions `json:"voteOptions"`
+	PrivateKeys   []string                   `json:"-"`
+	PublicKeys    []string                   `json:"-"`
+	QuestionIndex uint32                     `json:"questionIndex"`
+	CreationTime  time.Time                  `json:"creationTime"`
+	HaveResults   bool                       `json:"haveResults"`
+	FinalResults  bool                       `json:"finalResults"`
+}
+
+func (p Process) String() string {
+	b, _ := json.Marshal(p)
+	return string(b)
+}
+
+// Entity holds the db reference for an entity
+type Entity struct {
+	ID           types.HexBytes `badgerholdKey:"ID"`
+	CreationTime time.Time
+}
+
+// VotePackage represents the payload of a vote (usually base64 encoded)
+type VotePackage struct {
```

Comment:
This is not an scrutinizer specific type, that's actually a vochain type so it must not be here.